### PR TITLE
feat: add helm-repo release ci (with usage of github-pages as helm repo)

### PR DESCRIPTION
**Pull request by genofire**
* Resolves: #5371
* Target version: master 

### Summary
This PR adds an Github-Actions, on every change of the `version:` in `Chart.yaml` it will:
 1. bundle the helmchart to a `.tgz`
 2. add an git-tag / github-release with `<helmchartname>-<version>`:
 3. upload the bundle `.tgz` to the github-release
 4. update the `index.yaml` in branch `gh-pages`:
    - add the new helmchart release with url to the `tgz` on github-release
    - so this index.yaml should be accessible under: https://collaboraonline.github.io/online/index.yaml
 
---
so everybody could use it as an helm-repository like:
```bash
helm repo add collabora https://collaboraonline.github.io/online/
helm install --create-namespace --namespace collabora collabora-online collabora/collabora-online -f my_values.yaml
```

TODO:
 - [ ] create the branch `gh-pages` (an manuelle step on this repo)

### Checklist

- [x] Code is properly formatted
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by Rash419** [comment]
@genofire thanks for the PR :)

**Comment by Rash419** [comment]
@genofire if you have followed any tutorial , can you link it here ?

**Comment by genofire** [comment]
the used github action is from here:
https://github.com/helm/chart-releaser-action

this calls the tool:
https://github.com/helm/chart-releaser

**Comment by Rash419** [comment]
I am also confused, will it create index.yaml by itself ? Where will it store the file ? In gh-pages branch ?

**Comment by Rash419** [comment]
> I am confused, will it create index.yaml by itself or do we need to do that manually using `helm repo` ? Do we need to create tar balls manually ? Where will it store the file ? In gh-pages branch ?

answer:
This uses [@helm/chart-releaser-action](https://www.github.com/helm/chart-releaser-action) to turn your GitHub project into a self-hosted Helm chart repo. It does this – during every push to main – by checking each chart in your project, and whenever there's a new chart version, creates a corresponding [GitHub release](https://help.github.com/en/github/administering-a-repository/about-releases) named for the chart version, adds Helm chart artifacts to the release, and creates or updates an index.yaml file with metadata about those releases, which is then hosted on GitHub Pages. You do not need an index.yaml file in main at all because it is managed in the gh-pages branch.

File: .github/workflows/helm-release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        uses: azure/setup-helm@v3
+        with:
+          version: v3.12.1
+
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@v1.5.0
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+        with:
+          charts_dir: "helm/kubernetes"
```
Comment (Rash419) [inline .github/workflows/helm-release.yml:36]: this should be: 
`charts_dir: "kubernetes/helm"`

**Review by Rash419 — CHANGES_REQUESTED** [review]

**Comment by genofire** [comment]
so lets try / merge it

**Comment by genofire** [comment]
i also update the README, for using the new helm repo

**Comment by Rash419** [comment]
created empty gh-pages branch

**Review by Rash419 — APPROVED** [review]
Looks good to me!

**Comment by Rash419** [comment]
@genofire It says:
`Nothing to do. No chart changes detected.`

**Comment by Rash419** [comment]
@genofire do we need here only root files and `kubernetes/helm` folder ?
https://github.com/actions/checkout#fetch-only-the-root-files-and-github-and-src-folder

**Comment by Rash419** [comment]
> @genofire do we need here only root files and `kubernetes/helm` folder ? https://github.com/actions/checkout#fetch-only-the-root-files-and-github-and-src-folder

Answer: I don't think so

**Pull request merged**